### PR TITLE
Adds docs on how to enable DEBUG logging

### LINKED ISSUE #1944: Pulp 3 should have an environment variable (or another convenient way) to enable debug logging
Author: iballou (iballou)

Redmine Issue: 7812, https://pulp.plan.io/issues/7812

---

PULP_LOG_LEVEL=debug, or something like that would be nice.

### PR DESCRIPTION
**Pull request by bmbouter**
closes #1944

### CODE REVIEW

File: docs/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Enabling Debug Logging
+======================
+
+By default Pulp logs at INFO level, but enabling DEBUG logging can be a helpful thing to get more
+insight when things don't go as expected. This can be enabled with dynaconf using the examples
+below.
+
+Setting the following environment variable on one or more Pulp processes::
+
+    PULP_LOGGING={'dynaconf_merge': True, 'loggers': {'': {'handlers': ['console'], 'level': 'DEBUG'}}}
```
Comment (bmbouter) [inline docs/troubleshooting.rst:12]: @rochacbruno I tried maybe 5-6 variations to get this working and I wasn't able to. It doesn't seem to merge, it always overwrites. Got any pointers for me?

Comment (rochacbruno) [reply docs/troubleshooting.rst:12]: The challenge there is the empty string `''` being used as the dictionary key.
merge operates by key and the current implement is not able to work with empty/false keys.

If the logger was for example calles `root`

```py
PULP_LOGGING={'dynaconf_merge': True, 'loggers': {'root': {'handlers': ['console'], 'level': 'DEBUG'}}}
```

Or

```bash
PULP_LOGGING__loggers__root__level=DEBUG
```

I don't know right a workaround, we need changes on dynaconf to have some aditional syntax to work on empty keys.

Comment (bmbouter) [reply docs/troubleshooting.rst:12]: @rochacbruno thanks for the reply! Is there a bug or something I can follow? I'd like to document the limitation in a straightforward way for our users as it works towards a resolution.

Comment (rochacbruno) [reply docs/troubleshooting.rst:12]: @bmbouter this is not being tracked anywhere, the issue never happened before.

Comment (bmbouter) [reply docs/troubleshooting.rst:12]: @rochacbruno I added it here so I could at least reference the issue. I don't expect it to be fixed unless it's easy. https://github.com/rochacbruno/dynaconf/issues/747

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Due to `a bug in Dynaconf <https://github.com/rochacbruno/dynaconf/issues/747>`_, the following
+example won't work, but is included for illustrative purposes::
+
+    PULP_LOGGING={'dynaconf_merge': True, 'loggers': {'': {'handlers': ['console'], 'level': 'DEBUG'}}}
+
+.. tip::
+
+    As a workaround, you could specify the entire config with the `PULP_LOGGING` environment variable
+    and avoid using the "merge" feature from dynaconf. In that case you would specify
+    ``'level': 'DEBUG'`` in addition to your current config show with ``dynaconf list``.
```
Comment (ggainey) [inline docs/troubleshooting.rst:28]: ^show^shown

File: docs/troubleshooting.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+.. _enabling-debug-logging:
+
+Enabling Debug Logging
+----------------------
+
+By default Pulp logs at INFO level, but enabling DEBUG logging can be a helpful thing to get more
+insight when things don't go as expected. This can be enabled with dynaconf using the examples
+below.
+
+Designating a Python based settings file, and putting the DEBUG logging configuration there::
```
Comment (ggainey) [inline docs/troubleshooting.rst:14]: ^Python based^Python-based

File: docs/troubleshooting.rst [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+.. tip::
+
+    As a workaround, you could specify the entire config with the `PULP_LOGGING` environment variable
+    and avoid using the "merge" feature from dynaconf. In that case you would specify
+    ``'level': 'DEBUG'`` in addition to your current config show with ``dynaconf list``.
+
+Then when starting Pulp you should see a lot more information logged.
+
+To ensure you've enabled the settings correctly, view them with the ``dynaconf list`` command which
+you can read more about :ref:`viewing-settings`. If configured correctly you should see::
```
Comment (ggainey) [inline docs/troubleshooting.rst:33]: ^command which you can read more about:ref:`viewing-settings`.^command (for more information, see :ref:`viewing-settings`).

**Review by ggainey — CHANGES_REQUESTED** [review]
Just some minor wordsmithing tweaks, looks great otherwise!

**Comment by bmbouter** [comment]
@ggainey all pushed if you could re-review plz.

**Review by ggainey — APPROVED** [review]
:+1:

**Review by dkliban — COMMENTED** [review]
Should we include anything about setting 'DJANGO_SETTINGS_MODULE' before calling into dynaconf?

**Comment by bmbouter** [comment]
> Should we include anything about setting 'DJANGO_SETTINGS_MODULE' before calling into dynaconf?

Good idea, let me add that now.

**Review by dkliban — APPROVED** [review]

**Comment by curiositycasualty** [comment]
`PULP_LOGGING='@json {"dynaconf_merge": true, "loggers": {"": {"handlers": ["console"], "level": "DEBUG"}}}'`
~ https://github.com/rochacbruno/dynaconf/issues/747

**Pull request merged**